lib/ex_unit/lib/ex_unit/doc_test.ex

```diff
@@ -522,6 +522,7 @@ defmodule ExUnit.DocTest do
     rescue
       e ->
         ex_message = "(#{inspect(e.__struct__)}) #{Exception.message(e)}"
+        ex_message = String.replace(ex_message, ["\e[31m", "\e[0m"], "")
```

Comment:
Can we disable ansi instead for these tests?

Reply:
I've stripped any ansi characters here when formatting the exception because the ansi "reset" character inside the message conflicts with the doctest error message. Without this line it looks like this: 
![image](https://github.com/elixir-lang/elixir/assets/59743220/d6052d81-7c9b-4733-b9c7-7f8df1ab11f4)

With it, it looks like this: 
![image](https://github.com/elixir-lang/elixir/assets/59743220/b19b0cbf-824c-4683-bcf2-dd80b7da9a06)

I can write a test asserting that ex_unit error messages contain only one ansi red and ansi reset characters if you think it would be nice

We could also have a way of disabling ansi formatting in exceptions, but since it's only for two specific cases, I couldn't think of an ergonomic solution

Reply:
Maybe we should check if the exception has _any_ escape code and, if there is, we don't add any? I.e. we show the first line in default color?

Reply:
I like this idea, but I had some issues implementing it:
- It breaks the ex_unit's `colors: false` configuration if someone has ansi_enabled, so in this case I think we would need to strip the ansi characters anyway
- Even when setting `ansi_enabled` to false before the `ExUnit.run` call in the tests the exceptions still had ansi characters, but I'm still not sure why this happened

Reply:
Ping. :)

Reply:
Hi! I was still taking a look at this yesterday, I managed to get it working in the way you mentioned, but the I noticed it breaks the `colors: [enabled: false]` option of ex_unit.

We would would need to have a way to access the ex_unit options in the line that builds the message in order to remove the ansi characters if colors are disabled https://github.com/elixir-lang/elixir/blob/0cdd845004faf3cf0e7930d51e59fc01d9f07e2b/lib/ex_unit/lib/ex_unit/doc_test.ex#L524

If we strip them by default we won't have this problem as ex_unit will control the colors here.

Looking at the code, it seems that it would require a considerable refactor to do that, do you think it's worth for us to try it now or we could follow with the "all red" exception format for now?

Reply:
I wouldn't worry about. ExUnit colors configures the colors ExUnit emits, not the ones by code invoked by ExUnit.

Reply:
In other words, the strategy I proposed above is likely fine. Ship it :)

Reply:
Nice! Sorry for the back and forths, I think now we're good to go :)